data/src/main/java/com/linkedin/data/schema/annotation/ResolvedPropertiesReaderVisitor.java

```diff
@@ -122,8 +122,26 @@ public void callbackOnContext(DataSchemaRichContextTraverser.TraverserContext co
     return null;
   }
 
-  public Map<PathSpec, Map<String, Object>> getLeafFieldsPathSpecToResolvedPropertiesMap()
+  /**
+   * This method is deprecated and should not be used due to performance consideration, because this method will generate string and use that as map keys,
+   * and it is not necessarily memory-efficient.
+   *
+   * User should use {@link #getLeafFieldsPathSpecObjToResolvedPropertiesMap}, which use PathSpec object as map key.
+   * @return a map with {@link PathSpec} string points to leaf field as map key and the resolved properties as its value
+   *
+   */
+  @Deprecated
+  public Map<String, Map<String, Object>> getLeafFieldsPathSpecToResolvedPropertiesMap()
   {
-    return _leafFieldsPathSpecToResolvedPropertiesMap;
+    return _leafFieldsPathSpecObjToResolvedPropertiesMap.entrySet()
+                                                        .stream()
+                                                        .collect(Collectors.toMap(e -> e.getKey().toString(),
+                                                                                  Map.Entry::getValue));
+
+  }
+
+  public Map<PathSpec, Map<String, Object>> getLeafFieldsPathSpecObjToResolvedPropertiesMap()
```

Comment:
I understand that you need to rename the method to avoid conflicts, but this is very wordy. I'd suggest cutting it down. How about `getLeafFieldToResolvedPropertiesMap`? You don't need to say "pathSpec" because it's implied based on the return type.